### ISSUE TITLE
Caja should show more icons per page

### ISSUE THREAD

**Issue by MarioMey** [mate-desktop/caja#1313]
This is Caja showing Icons in 400%. Only 9 icons are shown.

![imagen](https://user-images.githubusercontent.com/4954109/67485523-99315e00-f640-11e9-873d-e05602299eaa.png)

I would like to see... 18 icons. There's place for everyone.

![imagen](https://user-images.githubusercontent.com/4954109/67485586-ba924a00-f640-11e9-8bea-3543b0461773.png)

#### MATE general version
1.20.4
#### Linux Distribution
Debian Buster

**Comment by MarioMey** [comment]
Sorry, I just found "Compact view".

**Comment by bart9h** [comment]
After reading this bug report, and finally found the `Edit` => `Preferences` => `Views` => `Icon View Defaults` => `Use compact layout` option.

After checking it, I still had to do `View` => `Reset View to defaults` to had the effect applied.

Maybe it should be easier to switch to the compact mode.

Apart from that, I think the spacing at 400% is way too big, even when not using the compact mode.

**Comment by lukefromdc** [comment]
While 400% zoomed is the limiting case, for the more normal icon sizes
I find that what Nautilusdid and Nemo eventually also applied createsa
very visually distracting view. Any changesthat are not
user-selectable are bound to createproblems for somebody, especially
as Caja is the last Nautilus-based file manager to offer the
traditional views and appearance.

On 4/19/2020 at 9:54 PM, "Rodolfo Borges"  wrote:
	After reading this bug report, and finally found the Edit =>
Preferences => Views => Icon View Defaults => Use compact layout
option. 

	After checking it, I still had to do View => Reset View to defaults
to had the effect applied. 

	Maybe it should be easier to switch to the compact mode. 

	Apart from that, I think the spacing at 400% is way too big, even
when not using the compact mode.

**Comment by linuxgirl22** [comment]
"After checking it, I still had to do View => Reset View to defaults"
Thanks @bart9h ... couldn't understand why usual Compact View resulted in such spaced out icons suddenly, but had no idea has to reset view. That worked!

**Comment by bart9h** [comment]
I think it is wrong to close this issue without fixing it.

The default behavior is still broken, even if there's a (somewhat difficult to find) workaround available.

**Comment by linuxgirl22** [comment]
I agree @bart9h, and thank you very much for reopening this @MarioMey.

**Comment by lukefromdc** [comment]
What I do NOT see happening is changing the as-shipped  default icon view to show more icons more closely spaced, with no way to change them back like Nautilus did. That kind of change in fact is why I no longer use Nemo for root file manager windows (as I used to do to distinguish them by the red warning bar). I am so opposed to forcing closer spacing (looks to cluttered to me) t that I would have to locally revert any such change in all of my builds of Caja if such a  commit were merged.  

NOT reclosing however as you mention a possible  problem getting your default view preference applied globally (as the default, to all caja navigation windows) and if that is doing something other than what it would be expected to do (not sure if it is), that is a bug needing a fix.

 Edit =>Preferences => Views => Icon View Defaults => Use compact layout should work going forward (be applied to all new windows), and you should only have to View => Reset View to defaults on existing windows.  Note than changing other view defaults (e.g browser vs spatial) also does not change existing windows, 

If this works and your default is applied going forward, we don't have an actual bug. We do NOT want to change windows that have been customized to the new default however, as that would cause changing the default to trash any existing per-folder choices users may have made and we don't want to do that.   A valid feature request would be and ask for a "apply to existing windows too? UI element. 

Making the icon view (as opposed to compact view) icon spacing user-configurable from the view menu should be possible, but a lot more work. I would have no problem merging such a PR if someone writes it and it works, but don't have time to write that one myself

**Comment by linuxgirl22** [comment]
@lukefromdc just requesting it works as normal, as something seems to have changed in recent weeks possibly? Not about obliterating others' settings or how Caja usually works.

**Comment by bart9h** [comment]
> I am so opposed to forcing closer spacing (looks to cluttered to me)

I agree with that. No need to make the Compact look the default.

Just reduce a bit of the horizontal spacing at 400%, it's just way too big.
More than half of the screen is blank space.

**Comment by Baerbeisser** [comment]
Even on 100% it's too big. Thunar shows 7 rows, while Caja only shows 5 rows in the same window size.

There's no dconf setting to change that and i can't find anything in gtk inspector to change via CSS. Something like Thunar's -`ThunarAbstractIconView-row-spacing` and `-ThunarAbstractIconView-column-spacing` would work too.

**Comment by Baerbeisser** [comment]
This is not about forcing things on anyone but to provide tools for people who *wish* to change things.

**Comment by lukefromdc** [comment]
Turns out in the caja->View->Arrange Items menuitem, we already have the "compact layout" option as well as having the entirely separate Compact View which is smaller and denser yet. The same menu is also accessable from the right-click menu brought up by right-clicking on any available space in the icon view. 

Thus this has already been implemented and is selectable. Just might not be all that well known. 

![Image](https://github.com/user-attachments/assets/1daf47b6-c8fb-46b4-a644-7f286ec7b831)

Thankfully, this is also shown in mate-user-guide's help for caja. If new users don't read the manual they won't find all of MATE's many features except by accident

![Image](https://github.com/user-attachments/assets/85e403ae-c3ab-42f9-bef7-95828b923666)

**Comment by Baerbeisser** [comment]
Ok, fair.

> I personally feel the compact layout does not look good-and that the compact layout in Nemo and Nautilus does not look good either to me. I am very used to the traditional spacing in Caja and anything else simply looks off to me though

Same for me with Thunar with a CSS fix, lol.
